openslides/agenda/models.py

```diff
@@ -330,13 +282,12 @@ def _calc_sibling_no(self):
         """
         Counts all siblings on the same level which are AGENDA_ITEMs.
         """
-        sibling_no = 0
-        prev_sibling = self.get_previous_sibling()
-        while prev_sibling is not None:
-            if prev_sibling.type == self.AGENDA_ITEM:
-                sibling_no += 1
-            prev_sibling = prev_sibling.get_previous_sibling()
-        return sibling_no + 1
+        siblings = (Item.objects.filter(parent=self.parent, type=self.AGENDA_ITEM)
+                        .order_by('weight'))
+        for index, item in enumerate(siblings):
+            if item == self:
+                return index + 1
+        raise RuntimeError("Item {} has to be a sibling of himself".format(self))
```

Comment:
I rewrote the hole method, gave it a new name and changed the docstring. Please have another look at it.